tools/testing/requirements.in

```diff
@@ -0,0 +1,5 @@
+coverage
```

Comment:
We're not picking specific versions any more?

Reply:
not in the `requirements.in` file unless we have a specific reason to - the `requirements.txt` file is generated from these and that does lock/pin/pick specific versions

that said we can set versions here if we need to and `pip-compile` should respect them (not sure about dependabot)

Reply:
some further investigation of using `pip-compile` 

one annoying issue is that it uses sets and doesnt order the hashes, which means running it is non-deterministic (ie the hashes keep reordering) - which is a pita 8/. I had thought that this problem was due to running different versions of pip-compile, but now it seems like the issue is more as described.

the other thing which doesnt work so well with pip-compile is conditional deps

if you set a conditional dep on eg `sys_platform = "win32"` or somesuch pip-compile respects it in the sense that if you are not on windows it doesnt get included in the requirements.txt file

this is pretty rubbish for mananging repo requirements if you expect different install/run environments

Reply:
this is already an issue as windows seems to be expecting `pyreadline` in some of my prs

we can just add it as dep, currently that would make linux pull it in unnecessarily, but that also could be resolved with (the new) pip_install_incremental in bazel

the problem with this approach occurs where there is a dep for some particular env that you really dont want to be pulled in, but im not sure that is an issue for envoy. In that case the best thing would be to separate the deps in bazel i think